chain/client/src/stateless_validation/state_witness_producer.rs

```diff
@@ -281,3 +288,37 @@ impl Client {
         Ok(source_receipt_proofs)
     }
 }
+
+// Add random data inside a dummy transaction to increase size of ChunkStateWitness.
+// Data is random to avoid compression getting in the way.
+// It doesn't have to be a cryptographically secure rng, just random enough to avoid compression.
+#[cfg(feature = "artificial_witness_size")]
+fn add_artificial_witness_size(
+    mut witness: ChunkStateWitness,
+    size_to_add: bytesize::ByteSize,
+) -> ChunkStateWitness {
+    use near_crypto::{KeyType, PublicKey, Signature};
+    use near_primitives::action::{Action, DeployContractAction};
+    use near_primitives::transaction::{SignedTransaction, Transaction, TransactionV1};
+    use rand::RngCore;
+    use std::str::FromStr;
+
+    let size_as_usize: usize = size_to_add.as_u64().try_into().expect("Can't convert u64 to usize");
+    let mut random_data = vec![0; size_as_usize];
```

Comment:
This can potentially add more than exactly requested number of bytes due to encoding overhead changing. Is that okay?

Reply:
> This can potentially add more than exactly requested number of bytes due to encoding overhead changing. Is that okay?

It's fine we're interested about the difference between 10 MB and 15 MB, a kilobyte of difference doesn't matter that much.